### PR TITLE
Python 3: Port tests in html/semantics/scripting-1

### PR DESCRIPTION
**Pull request by ziransun**

### CODE REVIEW

File: html/semantics/scripting-1/the-script-element/moving-between-documents/resources/moving-between-documents-iframe.py [hunk 1]
```diff
@@ -1,56 +1,59 @@
...
 
 def main(request, response):
-  inlineOrExternal = request.GET.first("inlineOrExternal", "null")
-  hasBlockingStylesheet = request.GET.first("hasBlockingStylesheet", "true") == "true"
-  result = request.GET.first("result", "success")
-  type = "text/javascript" if request.GET.first("type", "classic") == "classic" else "module"
+  inlineOrExternal = request.GET.first(b"inlineOrExternal", b"null")
+  hasBlockingStylesheet = request.GET.first(b"hasBlockingStylesheet", b"true") == b"true"
+  result = request.GET.first(b"result", b"success")
+  type = b"text/javascript" if request.GET.first(b"type", b"classic") == b"classic" else b"module"
```
Comment (jdm) [inline html/semantics/scripting-1/the-script-element/moving-between-documents/resources/moving-between-documents-iframe.py:15]: If we use the `u` prefix for the "text/javascript" or "module" values, we don't need to use isomorphic_decode on `type` later.

**Review by jdm — CHANGES_REQUESTED** [review]

File: html/semantics/scripting-1/the-script-element/moving-between-documents/resources/moving-between-documents-iframe.py (hunk 1 above)
Comment (ziransun) [reply html/semantics/scripting-1/the-script-element/moving-between-documents/resources/moving-between-documents-iframe.py]: Yes. Updated the code. Thank you!

**Comment by jdm** [comment]
The taskcluster tests that are failing don't appear to use any of the files modified by this PR. I wonder if this is a failure on the latest git revision?
```
 9:40.06 INFO ## Unstable results ##

 9:40.06 INFO |                                            Test                                           |                                                                                                                         Subtest                                                                                                                          |            Results            |                Messages                |
 9:40.06 INFO |-------------------------------------------------------------------------------------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|-------------------------------|----------------------------------------|
 9:40.06 INFO | `/fetch/metadata/redirect/redirect-https-downgrade.sub.html`                              | `Https downgrade font => No headers: sec-fetch-mode`                                                                                                                                                                                                     | **PASS: 1/10, MISSING: 9/10** |                                        |
 9:40.06 INFO | `/fetch/metadata/redirect/redirect-https-downgrade.sub.html`                              | `Https downgrade font => No headers: sec-fetch-site`                                                                                                                                                                                                     | **PASS: 1/10, MISSING: 9/10** |                                        |
 9:40.07 INFO | `/fetch/metadata/redirect/redirect-https-downgrade.sub.html`                              | `Https downgrade font => No headers: sec-fetch-user`                                                                                                                                                                                                     | **PASS: 1/10, MISSING: 9/10** |                                        |
 9:40.07 INFO | `/fetch/metadata/redirect/redirect-https-downgrade.sub.html`                              | `Https downgrade font => No headers: sec-fetch-dest`                                                                                                                                                                                                     | **PASS: 1/10, MISSING: 9/10** |                                        |
 9:40.07 INFO | `/webaudio/the-audio-api/the-mediaelementaudiosourcenode-interface/cors-check.https.html` | `  Recorded channel #0 is not constantly 0 (contains 35135 different values).`                                                                                                                                                                           | **PASS: 1/10, MISSING: 9/10** |                                        |
 9:40.07 INFO | `/webaudio/the-audio-api/the-mediaelementaudiosourcenode-interface/cors-check.https.html` | `  Recorded channel #1 is not constantly 0 (contains 35137 different values).`                                                                                                                                                                           | **PASS: 1/10, MISSING: 9/10** |                                        |
 9:40.07 INFO | `/webaudio/the-audio-api/the-mediaelementaudiosourcenode-interface/cors-check.https.html` | `  Recorded channel #2 is not constantly 0 (contains 35136 different values).`                                                                                                                                                                           | **PASS: 1/10, MISSING: 9/10** |                                        |
 9:40.07 INFO | `/webaudio/the-audio-api/the-mediaelementaudiosourcenode-interface/cors-check.https.html` | `  Recorded channel #3 is not constantly 0 (contains 35136 different values).`                                                                                                                                                                           | **PASS: 1/10, MISSING: 9/10** |                                        |
 9:40.07 INFO | `/webaudio/the-audio-api/the-mediaelementaudiosourcenode-interface/cors-check.https.html` | `  Recorded channel #0 is not constantly 0 (contains 44095 different values).`                                                                                                                                                                           | **PASS: 9/10, MISSING: 1/10** |                                        |
 9:40.07 INFO | `/webaudio/the-audio-api/the-mediaelementaudiosourcenode-interface/cors-check.https.html` | `  Recorded channel #1 is not constantly 0 (contains 44097 different values).`                                                                                                                                                                           | **PASS: 9/10, MISSING: 1/10** |                                        |
 9:40.07 INFO | `/webaudio/the-audio-api/the-mediaelementaudiosourcenode-interface/cors-check.https.html` | `  Recorded channel #2 is not constantly 0 (contains 44095 different values).`                                                                                                                                                                           | **PASS: 9/10, MISSING: 1/10** |                                        |
 9:40.07 INFO | `/webaudio/the-audio-api/the-mediaelementaudiosourcenode-interface/cors-check.https.html` | `  Recorded channel #3 is not constantly 0 (contains 44096 different values).`                                                                                                                                                                           | **PASS: 9/10, MISSING: 1/10** |                                        |
 9:40.07 INFO | `/webaudio/the-audio-api/the-mediaelementaudiosourcenode-interface/no-cors.https.html`    | `X Recorded channel #0: Expected 0 for all values but found 42815 unexpected values: \n\tIndex\tActual\n\t[1281]\t0.015655517578125\n\t[1282]\t0.031280517578125\n\t[1283]\t0.0467529296875\n\t[1284]\t0.06201171875\n\t...and 42811 more errors.`       | **FAIL: 1/10, MISSING: 9/10** | `assert_true: expected true got false` |
 9:40.07 INFO | `/webaudio/the-audio-api/the-mediaelementaudiosourcenode-interface/no-cors.https.html`    | `X Recorded channel #1: Expected 0 for all values but found 42817 unexpected values: \n\tIndex\tActual\n\t[1281]\t0.015655517578125\n\t[1282]\t0.031280517578125\n\t[1283]\t0.046722412109375\n\t[1284]\t0.06201171875\n\t...and 42813 more errors.`     | **FAIL: 1/10, MISSING: 9/10** | `assert_true: expected true got false` |
 9:40.07 INFO | `/webaudio/the-audio-api/the-mediaelementaudiosourcenode-interface/no-cors.https.html`    | `X Recorded channel #2: Expected 0 for all values but found 42815 unexpected values: \n\tIndex\tActual\n\t[1281]\t0.015655517578125\n\t[1282]\t0.031280517578125\n\t[1283]\t0.046722412109375\n\t[1284]\t0.062042236328125\n\t...and 42811 more errors.` | **FAIL: 1/10, MISSING: 9/10** | `assert_true: expected true got false` |
 9:40.07 INFO | `/webaudio/the-audio-api/the-mediaelementaudiosourcenode-interface/no-cors.https.html`    | `X Recorded channel #3: Expected 0 for all values but found 42816 unexpected values: \n\tIndex\tActual\n\t[1281]\t0.015655517578125\n\t[1282]\t0.031280517578125\n\t[1283]\t0.046722412109375\n\t[1284]\t0.06207275390625\n\t...and 42812 more errors.`  | **FAIL: 1/10, MISSING: 9/10** | `assert_true: expected true got false` |
 9:40.07 INFO | `/webaudio/the-audio-api/the-mediaelementaudiosourcenode-interface/no-cors.https.html`    | `X Recorded channel #0: Expected 0 for all values but found 44095 unexpected values: \n\tIndex\tActual\n\t[0]\t0.157379150390625\n\t[1]\t0.16925048828125\n\t[2]\t0.180511474609375\n\t[3]\t0.19085693359375\n\t...and 44091 more errors.`               | **FAIL: 8/10, MISSING: 2/10** | `assert_true: expected true got false` |
 9:40.07 INFO | `/webaudio/the-audio-api/the-mediaelementaudiosourcenode-interface/no-cors.https.html`    | `X Recorded channel #1: Expected 0 for all values but found 44097 unexpected values: \n\tIndex\tActual\n\t[0]\t0.15740966796875\n\t[1]\t0.16925048828125\n\t[2]\t0.18048095703125\n\t[3]\t0.190887451171875\n\t...and 44093 more errors.`                | **FAIL: 8/10, MISSING: 2/10** | `assert_true: expected true got false` |
 9:40.07 INFO | `/webaudio/the-audio-api/the-mediaelementaudiosourcenode-interface/no-cors.https.html`    | `X Recorded channel #2: Expected 0 for all values but found 44095 unexpected values: \n\tIndex\tActual\n\t[0]\t0.157440185546875\n\t[1]\t0.169189453125\n\t[2]\t0.1805419921875\n\t[3]\t0.19085693359375\n\t...and 44091 more errors.`                   | **FAIL: 8/10, MISSING: 2/10** | `assert_true: expected true got false` |
 9:40.07 INFO | `/webaudio/the-audio-api/the-mediaelementaudiosourcenode-interface/no-cors.https.html`    | `X Recorded channel #3: Expected 0 for all values but found 44096 unexpected values: \n\tIndex\tActual\n\t[0]\t0.15740966796875\n\t[1]\t0.169281005859375\n\t[2]\t0.180389404296875\n\t[3]\t0.19097900390625\n\t...and 44092 more errors.`               | **FAIL: 8/10, MISSING: 2/10** | `assert_true: expected true got false` |
 9:40.07 INFO | `/webaudio/the-audio-api/the-mediaelementaudiosourcenode-interface/no-cors.https.html`    | `X Recorded channel #0: Expected 0 for all values but found 40127 unexpected values: \n\tIndex\tActual\n\t[3969]\t0.015655517578125\n\t[3970]\t0.031280517578125\n\t[3971]\t0.0467529296875\n\t[3972]\t0.06201171875\n\t...and 40123 more errors.`       | **FAIL: 1/10, MISSING: 9/10** | `assert_true: expected true got false` |
 9:40.07 INFO | `/webaudio/the-audio-api/the-mediaelementaudiosourcenode-interface/no-cors.https.html`    | `X Recorded channel #1: Expected 0 for all values but found 40129 unexpected values: \n\tIndex\tActual\n\t[3969]\t0.015655517578125\n\t[3970]\t0.031280517578125\n\t[3971]\t0.046722412109375\n\t[3972]\t0.06201171875\n\t...and 40125 more errors.`     | **FAIL: 1/10, MISSING: 9/10** | `assert_true: expected true got false` |
 9:40.07 INFO | `/webaudio/the-audio-api/the-mediaelementaudiosourcenode-interface/no-cors.https.html`    | `X Recorded channel #2: Expected 0 for all values but found 40128 unexpected values: \n\tIndex\tActual\n\t[3969]\t0.015655517578125\n\t[3970]\t0.031280517578125\n\t[3971]\t0.046722412109375\n\t[3972]\t0.062042236328125\n\t...and 40124 more errors.` | **FAIL: 1/10, MISSING: 9/10** | `assert_true: expected true got false` |
 9:40.07 INFO | `/webaudio/the-audio-api/the-mediaelementaudiosourcenode-interface/no-cors.https.html`    | `X Recorded channel #3: Expected 0 for all values but found 40128 unexpected values: \n\tIndex\tActual\n\t[3969]\t0.015655517578125\n\t[3970]\t0.031280517578125\n\t[3971]\t0.046722412109375\n\t[3972]\t0.06207275390625\n\t...and 40124 more errors.`  | **FAIL: 1/10, MISSING: 9/10** | `assert_true: expected true got false` |
 9:40.07 INFO 
 9:40.07 INFO ::: Running tests in a loop 10 times : FAIL
```

**Comment by jdm** [comment]
Nevermind,  I forgot that fetch/api/resources/redirect.py is modified. Those failures make more sense now.

File: fetch/api/resources/redirect.py [hunk 2]
```diff
@@ -2,68 +2,70 @@
...
             scheme = urlparse(url).scheme
-            if scheme == "" or scheme == "http" or scheme == "https":
-                url += "&" if '?' in url else "?"
+            if scheme == u"" or scheme == u"http" or scheme == u"https":
+                url += u"&" if u'?' in url else u"?"
                 #keep url parameters in location
                 url_parameters = {}
                 for item in request.GET.items():
                     url_parameters[item[0]] = item[1][0]
                 url += urlencode(url_parameters)
```
Comment (jdm) [inline fetch/api/resources/redirect.py:51]: I wonder if urlencode gets messed by the byte strings from request.GET.items?

Comment (ziransun) [reply fetch/api/resources/redirect.py:51]: Python 3 (from version 3.2) urlencode query parameter supports bytes and string objects. I did a few printouts and the output looked reasonable.

**Comment by jdm** [comment]
I think this must be another situation where the tests are already flaky, since the changes in this PR look fine to me.

**Pull request merged**